File: power_DAC/power_DAC.cpp
#include <iostream>
using namespace std;
static float power(int a, int b)
{
    if(!b)return 1;
    float temp=power(a,b/2);
    if(b%2==0)return temp*temp;
    else
    {
        if(b>0)return temp*temp*a;
        else return temp*temp/a;
    }
}
int main()
{
    cout<<("%d^%d = %.2f",3,4,power(2,-2));
    return 0;
}

File: power_DAC/power_DAC.py
def power(a,b):
    if(not b):return 1
    
    temp = power(a,int(b/2))
    
    if(not b%2):return temp*temp
    
    else:
        if(b>0):return temp*temp*a
        else: return temp*temp/a
        
print(power(2,-2))
